queryfilter/tests/test_datetimefilter.py

```diff
@@ -0,0 +1,101 @@
+from __future__ import absolute_import
+
+import pytest
+import dateutil.parser
+from test_app.models import DatetimeFilterTestingModel
+from ..datetimefilter import DatetimeRangeFilter
+
+
+ordered_triple_test_date_set = (
+    ("2016-12-29T00:00:00+08:00", "2016-12-30T00:00:00+08:0", "2016-12-31T00:00:00+08:0"),  # in three day
+    ("2016-12-31T00:00:00+08:00", "2016-12-31T01:00:00+08:0", "2016-12-31T02:00:00+08:0"),  # in one day
+    ("2016-12-31T23:00:00+08:00", "2017-01-01T00:00:00-08:0", "2018-12-31T02:00:00+08:0"),  # across year
+)
+
+date_time_parametrize = pytest.mark.parametrize("dates, test_data, queryset",
+                                                zip(ordered_triple_test_date_set,
+                                                    ordered_triple_test_date_set,
+                                                    ordered_triple_test_date_set,
+                                                    ),
+                                                indirect=['test_data', 'queryset'])
+
+
+@pytest.fixture
+def test_data(request):
+    data = [{FIELD_NAME: date} for date in request.param]
+    return data
+
+
+@pytest.fixture
+def queryset(request):
+    data = test_data(request)
+
+    for datum in data:
+        DatetimeFilterTestingModel.objects.create(**{
+            FIELD_NAME: dateutil.parser.parse(datum[FIELD_NAME])
+        })
+
+    return DatetimeFilterTestingModel.objects.all()
+
+
+FIELD_NAME = 'datetime'
+
+
+@pytest.mark.django_db
+class TestDateRangeFilter(object):
+
+    @date_time_parametrize
+    def test_out_of_range(self, dates, test_data, queryset):
+
+        date = dates[0]
+        date = "1990" + date[4:]
```

Comment:
this line doesn't seems to represent "out_of_range" very clearly 🤔
how about 

```
earliest_qualified = dates[0]
earlier_than_qualified = "1990" + earlier_than_qualified[4:]
```

And maybe couple more case for only later&earlier cases?